node/network/statement-distribution/src/tests.rs

```diff
@@ -1842,6 +1842,324 @@ fn peer_cant_flood_with_large_statements() {
 	executor::block_on(future::join(test_fut, bg));
 }
 
+// This test addresses an issue when received knowledge is not updated on a
+// subsequent `Seconded` statements
+//
+// To trigger this issue, one should modify `circulate_statement` function
+// where there is a selection of lucky/unlucky peers to make all peers unlucky.
+// It could be done, for example, by avoiding calling to `peer_knowledge.send`:
+// `let peers_to_send: Vec<(PeerId, bool)> = peers_to_send` <--- here
+//
+// Secondly, the proposed change to update received knowledge of a peer in the
+// function `handle_incoming_message` must also be removed to reproduce the issue.
+// This is done after `active_head.check_useful_or_unknown` check where a statement
+// is useful but known.
+#[test]
+fn handle_multiple_seconded_statements() {
+	let hash_a = Hash::repeat_byte(1);
```

Comment:
nit: make it a bit more concise to use this throughout

```suggestion
	let relay_parent = Hash::repeat_byte(1);
```